ui/src/components/userProfile/UserProfileDesktop.js

```diff
@@ -0,0 +1,577 @@
+import {
+  Box,
+  Text,
+  Tabs,
+  TabList,
+  Tab,
+  TabPanels,
+  TabPanel,
+  FormControl,
+  FormLabel,
+  FormHelperText,
+  Input,
+  InputGroup,
+  Button,
+  Stack,
+  Center,
+  Switch,
+  InputRightAddon,
+  Select,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalFooter,
+  VStack,
+  ModalCloseButton,
+} from "@chakra-ui/react";
+import { CheckCircleIcon } from "@chakra-ui/icons";
+import { useSelector } from "react-redux";
+import { useEffect, useState } from "react";
+import { useForm, Controller } from "react-hook-form";
+import {
+  ministryTeamList,
+  lifegroupList,
+  districtList,
+  campusList,
+  lifestageList,
+  countryList,
+  regionList,
+} from "../helpers/lists";
+import {
+  settableDataFields,
+  userDataCleanup,
+  getUserDataRequest,
+  updateUserDataRequest,
+} from "../helpers/userInformationHelpers";
+
+const UserProfileDesktop = (props) => {
+  const { register, control, handleSubmit, setValue, formState } = useForm();
+  const { errors } = formState;
+  const user = useSelector((state) => state.user);
+  const [userData, setUserData] = useState(null);
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const onModalClose = (e) => {
+    setModalOpen(false);
+  };
+
+  const fetchUserData = async () => {
+    if (user.id) {
+      const { data, status } = await getUserDataRequest(user.id);
+
+      if (status === 200) {
+        setUserData(data[0]);
+        setUserInformationFields(data[0]);
+      }
+    }
+  };
+
+  const setUserInformationFields = (userData) => {
+    for (let key in userData) {
+      if (settableDataFields.includes(key)) {
```

Comment:
not sure if i'm understanding right, but if a user has "focus" as lifestage, this will not show in their profile page no? since the if just fails and the input field will show blank?

Reply:
it currently will still show Campus as not applicable. i didn't handle hiding that one field in that way